src/features/bivariate_color_manager/components/LegendWithMap/LegendWithMap.tsx

```diff
@@ -0,0 +1,74 @@
+import clsx from 'clsx';
+import { Text } from '@konturio/ui-kit';
+import { i18n } from '~core/localization';
+import { BivariateLegend as BivariateLegendComponent } from '~components/BivariateLegend/BivariateLegend';
+import { joinAndCapitalizeItems } from '~utils/common';
+import s from './LegendWithMap.module.css';
+import type { BivariateColorManagerDataValue } from '~features/bivariate_color_manager/atoms/bivariateColorManagerResource';
+import type { LayerSelectionInput } from '~features/bivariate_color_manager/atoms/bivariateColorManager';
+
+export type LegendWithMapProps = {
+  selectedData: BivariateColorManagerDataValue;
+  layersSelection: LayerSelectionInput;
+};
+
+export const LegendWithMap = ({
+  selectedData,
+  layersSelection,
+}: LegendWithMapProps) => {
+  const { legend, directions } = selectedData;
+  const { horizontal, vertical } = layersSelection;
+  const horizontalLabel = horizontal?.label;
+  const verticalLabel = vertical?.label;
+
+  const meta = {
+    hints: {
+      y: {
+        label: horizontalLabel,
+        direction: directions.horizontal,
+      },
+      x: {
+        label: verticalLabel,
+        direction: directions.vertical,
+      },
+    },
+  };
+
+  const renderXAxisLabel = (_bivAxis, rootClassName) => (
```

Comment:
would be cool to hang on types for arguments here

Reply:
will do that! 👍